pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -284,6 +306,60 @@ func (ctrl *Controller) itmsConfDeleted(_ interface{}) {
 	ctrl.imgQueue.Add("openshift-config")
 }
 
+func (ctrl *Controller) clusterImagePolicyAdded(_ interface{}) {
+	if !ctrl.sigstoreAPIEnabled() {
+		return
+	}
+	ctrl.imgQueue.Add("openshift-config")
+}
+
+func (ctrl *Controller) clusterImagePolicyUpdated(_, _ interface{}) {
+	if !ctrl.sigstoreAPIEnabled() {
+		return
+	}
+	ctrl.imgQueue.Add("openshift-config")
+}
+
+func (ctrl *Controller) clusterImagePolicyDeleted(_ interface{}) {
+	if !ctrl.sigstoreAPIEnabled() {
+		return
+	}
+	ctrl.imgQueue.Add("image-policy-deleted")
+}
+
+func (ctrl *Controller) sigstoreAPIEnabled() bool {
+	featureGates, err := ctrl.featureGateAccess.CurrentFeatureGates()
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("error getting current featuregates: %v", err))
+		return false
+	}
+	klog.Infof("imageverification sigstore FeatureGates: %v", featureGates.Enabled(apicfgv1.FeatureGateSigstoreImageVerification))
+	return featureGates.Enabled(apicfgv1.FeatureGateSigstoreImageVerification)
+}
+
+func (ctrl *Controller) enableClusterImagePolicyHandler() {
+	ctrl.configInformerFactory.Config().V1alpha1().ClusterImagePolicies().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
In the bootstrap e2e test logs the featuregate flipped from false to true. The below error would occur if not call this in the synchandler L874.  I think the syncHandler is processing the Image CR when the featuregate changes because we all handle these CRs in the same handler.  https://github.com/openshift/machine-config-operator/blob/3305cebfc87aac8aff75d46683ca2aa12546e955/pkg/controller/container-runtime-config/container_runtime_config_controller.go#L873-L876
```
goroutine 3753 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x29e9f80?})
	/go/src/github.com/openshift/machine-config-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:56 +0xcd
panic({0x2247ca0?, 0x3f809d0?})
	/usr/lib/golang/src/runtime/panic.go:914 +0x21f
github.com/openshift/machine-config-operator/pkg/controller/container-runtime-config.(*Controller).syncImageConfig(0xc000a96a80, {0x261fb84, 0x10})
	/go/src/github.com/openshift/machine-config-operator/pkg/controller/container-runtime-config/container_runtime_config_controller.go:872 +0x56f
github.com/openshift/machine-config-operator/pkg/controller/container-runtime-config.(*Controller).processNextImgWorkItem(0xc000a96a80)
	/go/src/github.com/openshift/machine-config-operator/pkg/controller/container-runtime-config/container_runtime_config_controller.go:459 +0xcc
```

Reply:
I didn’t mean that this should not happen in the sync loop[*] but that it should hot happen _repeatedly_.

```go
if !ctrl.addedPolicyObservers {
   … AddEventHandler…
   ctrl.addedPolicyObservers = true
}
```